modules/ln_glgui/primitives.scm

```diff
@@ -211,6 +211,79 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 (glgui-utf8-set! #t)
 
+;;; BEGIN CODE TRANSITION
+;;;
+;;; Reducing the computational complexity of font operations improves
+;;; rendering time.
+
+(define-type ln-ttf:font
+  macros: prefix: macro-
+  desc ;; for now the legacy description of a font as a assoc-list
+  char->desc-table
+  )
+
+(define (ln-ttf:font? obj) (macro-ln-ttf:font? obj))
+
+(define find-font/desc)
+
+(define (ln-ttf:font-ref font char) ;; -> glyph
+  (cond
+   ((macro-ln-ttf:font? font) ;; TBD: leave this as the only case
+    (table-ref (macro-ln-ttf:font-char->desc-table font) char #f))
+   (else (ln-ttf:font-ref (find-font font) char))))
+
+(define (make-ln-ttf:font/desc fnt)
+  (let ((font-table (list->table
+                     (let ((double-the-key (lambda (x) (cons (car x) x))))
+                       ;; TBD: get rid of the doubling - requires changes to call sites
+                       (map double-the-key fnt)))))
+    (macro-make-ln-ttf:font fnt font-table)))
+
+(define find-font
+  ;; TBD consider amount of fonts and decide on font cache data
+  ;; structure.  Now assq-list; good for rather short lists, which
+  ;; are in line with good user interface design.
+  (let ((by-desc '() #;(make-table hash: eq?-hash))
+        (assq assq))
+    (define (%find-font/desc fnt)
+      (let ((hit (assq fnt by-desc) #;(table-ref by-desc fnt #f)))
+        (if hit (cdr hit) #; hit
+            (let ((font (make-ln-ttf:font/desc fnt)))
+              #;(table-set! by-desc fnt font)
+              (set! by-desc (cons (cons fnt font) by-desc))
+              font))))
+    (define (find-font fnt)
+      (cond
+       ((macro-ln-ttf:font? fnt) ;; new style
+        ;;
+        ;; TBD: after transition phase warn if this case is hit.
+        fnt)
+       ((and (pair? fnt) (pair? (car fnt))) ;; likely a legacy font
+        (%find-font/desc fnt))
+       (else (error "illegal font" fnt))))
+    (set! find-font/desc %find-font/desc)
+    find-font))
+
+(define (install-font-cache-backward-compatible-override!)
+  (let ((assoc.orig assoc))
+    ;; Backward compatibility: try to transparently hook into legacy
+    ;; lookups.
+    ;;
+    ;; TBD: 2020-08-23 This is an intermediate workaround to be removed
+    ;; ASAP.
+    (define (transparent-font-assoc k coll)
+      (cond
+       ((macro-ln-ttf:font? coll)
+        ;; TBD: after experimental phase warn if this case is hit.
+        (ln-ttf:font-ref coll k))
+       (else
+        ;; TBD: after transition phase error out if this case is hit.
+        (assoc.orig k coll))))
+    (set! assoc transparent-font-assoc)))
```

Comment:
This overrides assoc globally, right? We use assoc in other places so what kind of errors can we expect from this?